OpenRA.Mods.Common/Traits/Render/RenderDebugState.cs

```diff
@@ -65,7 +65,7 @@ public void OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
 
 		Color GetColor()
 		{
-			return self.EffectiveOwner != null ? self.EffectiveOwner.Owner.Color.RGB : self.Owner.Color.RGB;
+			return self.EffectiveOwner != null && self.IsInWorld ? self.EffectiveOwner.Owner.Color.RGB : self.Owner.Color.RGB;
```

Comment:
What are you actually trying to avoid here?  `self.EffectiveOwner.Owner.Color.RGB` should have nothing to do with whether the actor is in the world or not.